public/app/features/dashboard-scene/scene/layout-rows/RowItem.tsx

```diff
@@ -113,22 +113,24 @@ export class RowItem extends SceneObjectBase<RowItemState> implements LayoutPare
         )}
         ref={ref}
       >
-        <div className={styles.rowHeader}>
-          <button
-            onClick={model.onCollapseToggle}
-            className={styles.rowTitleButton}
-            aria-label={isCollapsed ? 'Expand row' : 'Collapse row'}
-            data-testid={selectors.components.DashboardRow.title(titleInterpolated)}
-          >
-            <Icon name={isCollapsed ? 'angle-right' : 'angle-down'} />
-            <span className={styles.rowTitle} role="heading">
-              {titleInterpolated}
-            </span>
-          </button>
-          {isEditing && (
-            <Button icon="pen" variant="secondary" size="sm" fill="text" onPointerDown={(evt) => onSelect?.(evt)} />
-          )}
-        </div>
+        {(!isHeaderHidden || showHiddenElements) && (
```

Comment:
And have something like this here
```suggestion
        {(!isHeaderHidden || (isEditing && showHiddenElements)) && (
```

Reply:
Merged this as an additional check to avoid future issues in case somebody removes `showHiddenElements: false` when exiting edit mode.